core/web/resolver/testdata/config-full.toml

```
@@ -35,14 +35,18 @@ LeaseRefreshInterval = '1s'
 [TelemetryIngress]
 UniConn = true
 Logging = true
-ServerPubKey = 'test-pub-key'
-URL = 'https://prom.test'
 BufferSize = 1234
 MaxBatchSize = 4321
 SendInterval = '1m0s'
 SendTimeout = '5s'
 UseBatchSend = true
 
+[[TelemetryIngressEndpoint]]
+Network = 'EVM'
+ChainID = '1'
+URL = 'https://prom.test'
+ServerPubKey = 'test-pub-key'
```

I think that using map types could allow us to express these more simply:
```suggestion
[[TelemetryIngressEndpoint.EVM.1]]
URL = 'https://prom.test'
ServerPubKey = 'test-pub-key'
```

---

I figured out how to do it with tables, but I was trying to unmarshal in the wrong type, I needed something like `map[string]map[string]telemetryIngressEndpoint`, doing it like this works but has some side effects, like when we are unmarshalling the config it will generate something like this:
```toml
[TelemetryIngressEndpoint]
[TelemetryIngressEndpoint.EVM]
[TelemetryIngressEndpoint.EVM.1]
URL = 'https://prom.test/'
ServerPubKey = 'test-pub-key'
```
I moved the config inside the `TelemetryIngress.Endpoints` field and now we have something like this:
```toml
[[TelemetryIngress.Endpoints]]
Network = 'EVM'
ChainID = '1'
URL = 'https://endpoint-1.test'
ServerPubKey = 'test-pub-key-1'
```